src/pages/DirectMessage.tsx

```diff
@@ -0,0 +1,252 @@
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "flex-end",
+  },
+  sentMessageText: {
+    padding: 1,
+    borderRadius: 10,
+    backgroundColor: "#22AA6F",
+  },
+};
+
+export type Message = {
+  messageId: number;
+  timestamp: string;
+  textContent: string;
+  sentUserId: number;
+  receivedUserId: number;
+};
+
+const DirectMessage = () => {
+  const { userId1, userId2 } = useParams();
+  const [textContent, setTextContent] = useState("");
+  const [messages, setMessages] = useState<Message[]>([]);
+  const user = useAppSelector((state) => state.user);
+  const { selectedConversation } = useAppSelector((state) => state.messages);
+  const messageRef = useRef<HTMLDivElement>(null);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    const fetchDirectMessage = async () => {
+      const result = await axios.get(
+        `http://localhost:3001/api/messages/${userId1}/${userId2}`
+      );
+      setMessages(result.data.messages as Message[]);
+      dispatch(
+        setSelectedConversation({
+          ...result.data.otherUser,
+          userId: Number(userId2),
+        })
+      );
+    };
+    fetchDirectMessage();
+  }, [dispatch, userId1, userId2]);
+
+  useEffect(() => {
+    // TODO: See if this is the best way to scroll to the bottom, and check edge cases
+    if (messageRef.current) {
+      messageRef.current.scrollTo(0, messageRef.current.scrollHeight);
+    }
+  }, [messages]);
+
+  const onSubmit = async (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    try {
+      const newMessage = (
+        await axios.post("http://localhost:3001/api/messages", {
+          sentUserId: user.userId,
+          receivedUserId: selectedConversation.userId,
+          textContent,
+        })
+      ).data as Message;
+      setTextContent("");
+      setMessages([...messages, newMessage]);
+      dispatch(
+        updateConversation({
+          displayName: selectedConversation.displayName,
+          otherUserId: selectedConversation.userId,
+          textContent: newMessage.textContent,
+          timestamp: newMessage.timestamp,
+          username: selectedConversation.username,
+        })
+      );
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  return (
+    <Stack direction="row" sx={styles.root}>
+      <ConversationList />
+      <Divider flexItem orientation="vertical" />
+      <Box sx={styles.directMessageActivityContainer}>
+        <Box sx={styles.directMessageActivityHeader}>
+          <Box sx={{ alignItems: "center", display: "flex" }}>
+            <Box sx={{ padding: 1 }}>
+              <Avatar />
+            </Box>
+            <Box sx={{ padding: 1 }}>
+              <Typography variant="subtitle1">
+                {selectedConversation.displayName}
+              </Typography>
+              <Typography variant="subtitle2">{`@${selectedConversation.username}`}</Typography>
+            </Box>
+          </Box>
+          <IconButton>
```

Comment:
for some reason the icon has an odd oval hover shape. I think we should remove the ripple effect

```suggestion
          <IconButton disableRipple>
```

Reply:
#53 should take care of this too